src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -92,7 +99,9 @@ protected void prepareRunInternal(BatchSinkContext context, BigQuery bigQuery, S
       }
 
       try {
-        Schema tableSchema = Schema.parseJson(argument.getValue());
+        Schema tableSchema = overrideOutputSchemaWithTableSchemaIfNeeded(
+            Schema.parseJson(argument.getValue()), tableName, collector);
```

Comment:
There doesn't appear to be any schema validation happening here. This can lead to situations where the pipeline is configured incorrectly, trying to write to a table with a completely different schema. We would use the table's schema instead of the data's schema, which would likely lead to weird null pointer or class cast exceptions that would be very difficult for the user to understand.

Instead of having this method lookup the table schema, I think it would be better to look it up outside and validate that it is compatible before passing it in. This would also avoid an extra schema lookup in the normal BQ sink.

Reply:
Schema validation added in prepareRunValidation(). Formatting fixed.